src/ucp/core/ucp_worker.c

```diff
@@ -584,21 +584,27 @@ ucp_worker_iface_handle_uct_ep_failure(ucp_ep_h ucp_ep, ucp_lane_index_t lane,
     ucp_worker_h worker = ucp_ep->worker;
     ucp_wireup_ep_t *wireup_ep;
 
-    /* If the failure happened on AUX EP of CM lane on a server EP,
-     * it means that client closed its CM_WIREUP_EP/AUX_EP and it
-     * was detected before receiving WIREUP_MSG/ACK from a client or
-     * marking a server's EP as REMOTE_CONNECTED was scheduled on a
-     * progress, but not completed yet (CM_WIREUP_EP/AUX_EP is
-     * closed when moving an EP to REMOTE_CONNECTED state) */
+    /* If UCP EP isn't failed and the failure happened on AUX EP of CM lane:
+     * - on a server EP:
+     *   it means that client closed its CM_WIREUP_EP/AUX_EP and it was detected
+     *   before receiving WIREUP_MSG/ACK from a client or marking a server's EP
+     *   as REMOTE_CONNECTED was scheduled on a progress, but not completed yet
+     *   (CM_WIREUP_EP/AUX_EP is closed when moving an EP to REMOTE_CONNECTED
+     *   state)
+     * - on a client EP:
+     *   it means that server closed its CM_WIREUP_EP/AUX_EP and it was detected
+     *   when marking a client's EP as REMOTE_CONNECTED was scheduled on a
+     *   progress, but not completed yet (CM_WIREUP_EP/AUX_EP is closed when
+     *   moving an EP to REMOTE_CONNECTED state)
+     * So, no need to invoke the error handling flow for UCP EP to not fail
+     * normal connection establishment flow. If the real error was detected by
+     * AUX EP, other lanes will be failed too later. */
     wireup_ep = ucp_wireup_ep(ucp_ep->uct_eps[lane]);
-    if ((lane == ucp_ep_get_cm_lane(ucp_ep))         &&
+    if (!(ucp_ep->flags & UCP_EP_FLAG_FAILED) &&
+        (lane == ucp_ep_get_cm_lane(ucp_ep)) &&
         (lane == ucp_ep_get_wireup_msg_lane(ucp_ep)) &&
-        (wireup_ep != NULL)                          &&
+        (wireup_ep != NULL) &&
         ucp_wireup_aux_ep_is_owner(wireup_ep, uct_ep)) {
-        ucs_assert(ucp_ep->flags & UCP_EP_FLAG_CONNECT_PRE_REQ_QUEUED);
```

Comment:
can you please elaborate when this assumption is incorrect?

Reply:
I've added elaboration on in the comment above
e.g. on a client side - when marking UCP EP as REMOTE_CONNECTED is in progress, but a server received WIREUP_MSG/ACK and closed its AUX EP, so failure was detected